tests/resolvers/Mutation/togglePinnedPost.spec.ts

```diff
@@ -0,0 +1,217 @@
+    const { requestContext } = await import("../../../src/libraries");
+    vi.spyOn(requestContext, "translate").mockImplementationOnce(
+      (message) => `Translated ${message}`
+    );
+
+    try {
+      const args: MutationTogglePinnedPostArgs = {
+        id: Types.ObjectId().toString(),
+      };
+
+      vi.doMock("../../../src/constants", async () => {
+        const actualConstants: object = await vi.importActual(
+          "../../../src/constants"
+        );
+        return {
+          ...actualConstants,
+          IN_PRODUCTION: true,
+        };
+      });
+
+      const context = {
+        userId: testUser!._id,
+      };
+
+      const { togglePinnedPost: togglePinnedPostResolver } = await import(
+        "../../../src/resolvers/Mutation/togglePinnedPost"
+      );
+
+      await togglePinnedPostResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(error.message).toEqual(`Translated ${POST_NOT_FOUND_MESSAGE}`);
+    }
+  });
+
+  it(`throws NotAuthorized error if the user is not the admin of the org or a superadmin`, async () => {
+    const { requestContext } = await import("../../../src/libraries");
+    const spy = vi
+      .spyOn(requestContext, "translate")
+      .mockImplementationOnce((message) => `Translated ${message}`);
+    try {
+      const args: MutationTogglePinnedPostArgs = {
+        id: testPost!.id,
+      };
+
+      const context = {
+        userId: randomUser!._id,
+      };
+
+      vi.doMock("../../../src/constants", async () => {
+        const actualConstants: object = await vi.importActual(
+          "../../../src/constants"
+        );
+        return {
+          ...actualConstants,
+          IN_PRODUCTION: true,
+        };
+      });
+
+      const { togglePinnedPost: togglePinnedPostResolver } = await import(
+        "../../../src/resolvers/Mutation/togglePinnedPost"
+      );
+
+      await togglePinnedPostResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(spy).toBeCalledWith(USER_NOT_AUTHORIZED_TO_PIN.message);
+      expect(error.message).toEqual(
+        `Translated ${USER_NOT_AUTHORIZED_TO_PIN.message}`
+      );
+    }
+  });
+
+  it(`adds a post to the pinnedPosts for an org`, async () => {
+    const { requestContext } = await import("../../../src/libraries");
+    vi.spyOn(requestContext, "translate").mockImplementationOnce(
+      (message) => `Translated ${message}`
+    );
+    const args: MutationTogglePinnedPostArgs = {
+      id: testPost!._id,
+    };
+
+    const context = {
+      userId: testUser!._id,
+    };
+
+    const { togglePinnedPost: togglePinnedPostResolver } = await import(
+      "../../../src/resolvers/Mutation/togglePinnedPost"
+    );
+
+    await togglePinnedPostResolver?.({}, args, context);
+    const organization = await Organization.findOne({
+      _id: testPost!.organization,
+    }).lean();
+
+    const currentPostIsPinned = organization!.pinnedPosts.some(
```

Comment:
You also have to check that on the actual post, the boolean is toggled.

Reply:
Updated the tests